jcl/src/java.base/share/classes/openj9/tools/attach/diagnostics/base/DiagnosticUtils.java

```diff
@@ -120,6 +145,56 @@ private static DiagnosticProperties getHeapStatistics(String diagnosticCommand)
 		}
 		return DiagnosticProperties.makeStringResult(result);
 	}
+	
+	private static DiagnosticProperties getThreadInfo(String diagnosticCommand) {
+		DiagnosticProperties result = null;
+		boolean okay = true;
+		boolean addSynchronizers = false;
+		String[] parts = diagnosticCommand.split(DIAGNOSTICS_OPTION_SEPARATOR);
+		/* only one option ("-l[=<BOOLEAN>]") is allowed, so the diagnosticCommand can comprise at most
+		 * the base command and one option, with an optional value */
+		if (parts.length > 2) {
+			okay = false;
+		} else if (parts.length == 2) {
+			String option = parts[1];
+			if (option.startsWith(THREAD_LOCKED_SYNCHRONIZERS_OPTION)) {
+				if ((THREAD_LOCKED_SYNCHRONIZERS_OPTION.length() == option.length()) /* exact match */
+						|| option.toLowerCase().equals(THREAD_LOCKED_SYNCHRONIZERS_OPTION + "=true")) { //$NON-NLS-1$
+					addSynchronizers = true;
+				}
+			} else {
+				okay = false;
+			}
+		}
+		if (!okay) {
+			result = DiagnosticProperties.makeErrorProperties("Command not recognized: " + diagnosticCommand); //$NON-NLS-1$
+		} else {
+			StringWriter buffer = new StringWriter(2000);
+			PrintWriter bufferPrinter = new PrintWriter(buffer);
+			bufferPrinter.println(System.getProperty("java.vm.info")); //$NON-NLS-1$
+			bufferPrinter.println();
+			ThreadInfoBase[] threadInfos = dumpAllThreadsImpl(true, addSynchronizers, Integer.MAX_VALUE);
+			for (ThreadInfoBase ti : threadInfos) {
+				bufferPrinter.print(ti.toString());
+				if (addSynchronizers) {
+					LockInfoBase[] syncs = ti.getLockedSynchronizers();
+					bufferPrinter.printf("%n\tLocked ownable synchronizers: %d%n", //$NON-NLS-1$
+							Integer.valueOf(syncs.length));
+					for (LockInfoBase s : syncs) {
```

Comment:
I'd still like to see better local variable names than `ti`, `syncs` and `s`. I suggest `threadInfo`, `locks` and `lock`, respectively.

Reply:
Good idea.  I made the names more explicit: `currentThreadInfoBase`, `lockedSynchronizers`, and `currentLockedSynchronizer`.